### PR TITLE
8051: strict timestamp for 'servedAt'

### PR DESCRIPTION
**Pull request by kkoskelin**
This effort always converts YYYY-MM-DD to strict ISO before storage, and should do the same for any comparisons.

We push a date like "2020-01-02" through `createISODateString()` to produce "2020-01-02T05:00:00.000Z".

* This PR also contains a very important bugfix in a `DateHandler` function that was used to decide whether or not two dates represented the same calendar day.

### CODE REVIEW

**Comment by kkoskelin** [comment]
Closing, putting this effort on hold for now.

File: web-api/migration-terraform/main/lambdas/migrations/0025-docket-entry-received-at-strict-timestamp.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    if (item.pk.startsWith('case|') && item.sk.startsWith('docket-entry|')) {
+      if (item.receivedAt) {
+        item.receivedAt = createISODateAtStartOfDayEST(item.receivedAt);
+      }
+
+      new DocketEntry(item, { applicationContext }).validate();
+
+      applicationContext.logger.info('Updating receivedAt for docketEntry', {
+        pk: item.pk,
+        processingStatus: item.processingStatus,
```
Comment (matthopson) [inline web-api/migration-terraform/main/lambdas/migrations/0025-docket-entry-received-at-strict-timestamp.js]: Can probably remove this line

**Review by matthopson — APPROVED** [review]

**Review by matthopson — APPROVED** [review]

**Comment by kkoskelin** [comment]
PR workflow now at https://github.com/ustaxcourt/ef-cms/pull/969

**Comment by matthopson** [comment]
Closing in favor of https://github.com/ustaxcourt/ef-cms/pull/969

**Pull request closed without merging**